pkg/virt-config/config-map.go

```diff
@@ -507,6 +556,10 @@ func (c *ClusterConfig) GetConfig() (config *v1.KubeVirtConfiguration) {
 	} else {
 		err = setConfigFromKubeVirt(config, kv)
 	}
+	if err == nil {
+		err = setConfigFromCPUArch(config)
```

Comment:
It would be great if `setConfigFromCPUArch` could move insode `defaultClusterConfig`. I would give the `ClusterConfig` the arch field instead of the config itself and pass it to `defaultClusterConfig`. That moves the arch detection nicely to the outer layers.

Reply:
Hi Roman, I get your idea on setting CPUArch in defaultClusterConfig. 

But for other configurable values like `OVMFPath, MachineType, EmulatedMachines`, if we set them in `defaultClusterConfig`, when we only pass `CPUArch` to function `NewFakeClusterConfig`, these configurable values would not change.

take a test as an example, the test would be failed
```
        table.DescribeTable(" when machineType", func(cpuArch string, machineKey string, result string) {
                clusterConfig, _, _, _ := testutils.NewFakeClusterConfig(&kubev1.ConfigMap{
                        Data: map[string]string{
                                virtconfig.CPUArchKey:     cpuArch,
                                virtconfig.MachineTypeKey: machineKey,
                        },
                })
                fmt.Printf("Haolin test machineType cpuArch: %v\n", cpuArch)
                Expect(clusterConfig.GetMachineType()).To(Equal(result))
        },
                table.Entry("when set, GetMachineType should return the value", "", "pc-q35-3.0", "pc-q35-3.0"),
                // The following three test would failed
                table.Entry("when unset, GetMachineType should return the default", "amd64", "", virtconfig.DefaultAMD64MachineType),
                table.Entry("when unset, GetMachineType should return the default", "arm64", "", virtconfig.DefaultAARCH64MachineType),
                table.Entry("when unset, GetMachineType should return the default", "ppc64le", "", virtconfig.DefaultPPC64LEMachineType),
        )

```

Reply:
I will give it a try first